src/test.c

```diff
@@ -613,12 +609,23 @@ void test_hash_armv8_crypto_multiple_blocks() {
 }
 #endif
 
+int override_called;
+void test_override(unsigned char*, const unsigned char*, uint64_t) {
+    override_called += 1;
+}
+
+void test_hashtree_init() {
+    hashtree_init(test_override);
+
+    unsigned char digest[960];
+    hashtree_hash(digest, test_32_block, 30);
+    TEST_CHECK(override_called);
```

Comment:
`override_called` is unitilialized so this test can give false positivies. Better to initialize it to zero before and also add a `TEST_CHECK(!override_called);` before the call to `hashtree_init`

Reply:
should be fine: https://stackoverflow.com/questions/16015656/are-global-variables-always-initialized-to-zero-in-c

Reply:
check added: a4d60e8d7227b6e1b85446a7d471067960e464fb

Reply:
Oh right! I missed this.